File: client/src/lib/redux/apis/SuggestedQuestionApi.ts
import { SuggestedQuestionParams } from "../module";
import axiosClient from "./AxiosClient";

const SuggestedQuestionApi = {
  getSuggestedQuestions: (params?: SuggestedQuestionParams) => {
    return axiosClient.get(
      `/suggestedQuestions${
        params?.pageNumber ? `?pageNumber=${params?.pageNumber}` : ""
      }${params?.pageSize ? `&pageSize=${params?.pageSize}` : ""}`
    );
  },
};

export default SuggestedQuestionApi;

File: server/src/main/java/com/nva/server/apis/SuggestedQuestionApi.java
package com.nva.server.apis;

import com.nva.server.entities.SuggestedQuestion;
import com.nva.server.services.SuggestedQuestionService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/v1/suggestedQuestions")
@CrossOrigin
public class SuggestedQuestionApi {
    @Autowired
    private SuggestedQuestionService suggestedQuestionService;

    @GetMapping
    public ResponseEntity<List<SuggestedQuestion>> getSuggestedQuestions(@RequestParam Map<String, Object> params) {
        return ResponseEntity.ok(suggestedQuestionService.getSuggestedQuestionsV2(params));
    }
}
